### PR TITLE
Throw exception if result is not bool

### LINKED ISSUE #12: Get no logs.
Hello,

I have this in my code:

```php
        $logger = new Logger($subdomain);
        $logger->pushHandler(new LogzIoHandler('TOKEN'));
```

This is my region:

```php
Region eu-west-2
```

But I get no logs? Did I'm missing something?

### PR DESCRIPTION
**Pull request by OskarStark**
I didn't receive an error, but after adding this code I received:
CleanShot 2022-03-31 at 10 14 22@2x

### CODE REVIEW

File: src/Handler/LogzIoHandler.php [hunk 1]
```diff
@@ -80,7 +79,11 @@ protected function send($data): void
         curl_setopt($handle, CURLOPT_HTTPHEADER, ['Content-Type: application/json']);
         curl_setopt($handle, CURLOPT_RETURNTRANSFER, true);
 
-        Util::execute($handle);
+        $result = Util::execute($handle);
+        
+        if (is_string($result)) {
```
Comment (OskarStark) [inline src/Handler/LogzIoHandler.php]:
```suggestion
        if (is_string($result) && trim($result) !== '') {
```

**Comment by Chrico** [comment]
Hm..this looks kind of wrong to me. The [`Monolog\Handler\Curl\Util`](https://github.com/Seldaek/monolog/blob/main/src/Monolog/Handler/Curl/Util.php#L66) does return either a boolean ( `false` ) or the `$curlResponse` which could be a string from [`curl_exec()`](https://www.php.net/manual/en/function.curl-exec.php). Since we're setting `CURLOPT_RETURNTRANSFER ` to `true` in [LogzIoHandler.php#L819](https://github.com/inpsyde/logzio-monolog/blob/master/src/Handler/LogzIoHandler.php#L81), it will return the result as a string **on success**. 🤔 

> Returns true on success or false on failure. However, if the CURLOPT_RETURNTRANSFER option is [set](https://www.php.net/manual/en/function.curl-setopt.php), it will return the result on success, false on failure.

I'm wondering what the response in your case for `curl_getinfo()` contains. How does your configuration look like? Can you provide some code so that we can see how you instantiate everything?

---

**Edit:** Here are docs regarding "JSON uploads": https://docs.logz.io/shipping/log-sources/json-uploads.html#possible-responses - looks like following:

> 401 UNAUTHORIZED
> The token query string parameter is missing or not valid. Make sure you’re using the right account token.
> In the response body, you’ll see either “Logging token is missing” or “Logging token is not valid” as the reason for the response.

I guess it's saver to check the HTTP `curl_getinfo($ch, CURLINFO_HTTP_CODE)` to be `>= 400` to detect an error. Currently on success (Status Code 200) they return an empty string. But this can change in future.

We should also add some test cases for this. :)

**Comment by OskarStark** [comment]
>  it will return the result as a string on success. 🤔

Hmm, I didn't receive a string on success 🤔 , but indeed, the return is `false|string`.
However, this was the only way to come across the error message, maybe we should check the return status code there to be able to throw an exception?

> I guess it's saver to check the HTTP curl_getinfo($ch, CURLINFO_HTTP_CODE) to be >= 400 to detect an error. Currently on success (Status Code 200) they return an empty string. But this can change in future.

Oh I read this part right after writing my sentence :-D

**Comment by OskarStark** [comment]
It could also be helpful to add a `bool $debug = false` to (1) and enable (2) if true:
CleanShot 2022-03-31 at 11 14 06@2x

**Pull request closed without merging**